src/gcp/auth.ts

```diff
@@ -165,3 +165,42 @@ export async function disableUser(
   });
   return res.status === 200;
 }
+
+/**
+ * setCustomClaim sets a new custom claim on the uid specified in the project.
+ * @param project project identifier.
+ * @param uid the user id of the user from the firebase project.
+ * @param claim the key value in the custom claim.
+ * @param value the value in the custom claim.
+ * @return the results of the accounts update request.
+ */
+export async function setCustomClaim(
+  project: string,
+  uid: string,
+  claim: string,
+  value: string | number | boolean,
+): Promise<UserInfo> {
```

Comment:
At the API layer I'd prefer this be:

```ts
setCustomClaims(project: string, uid: string, claims?: Record<string,unknown>, options?: {merge?: boolean})
```

It's fine if the tool only sets one claim at a time, but if the underlying API supports multiple let's do multiple.

Also I believe custom claims can be more than just scalar values, e.g. `{roles: ["admin", "moderator"]}`

Reply:
What should merge do? Should it overwrite all existing claims or should it change the value of:

`roles: admin` to `roles: [admin, moderator]`?

Reply:
Simple merge. So if {merge: true} you do `{...oldValues, newValues}` and if `{merge: false}` you do `{newValues}` (overwrite all claims).

The tool should always merge.

Reply:
Done!